File: frontend/src/objects/order/Order.ts
import type { MerchantId, OrderId, RiderId, UserId, OrderStatus, RefundStatus } from '@/objects/shared/ids'
import type { Promotion } from '@/objects/shared/Promotion'
import type { Voucher } from '@/objects/shared/Voucher'

import type { OrderItem } from './OrderItem'
import type { OrderPriceBreakdown } from './OrderPriceBreakdown'
import type { OrderPriceSnapshot } from './OrderPriceSnapshot'
import type { OrderTimelineEvent } from './OrderTimelineEvent'

export interface Order {
  id: OrderId
  customerId: UserId
  customerName: string
  customerPhone: string
  merchantId: MerchantId
  riderId?: RiderId
  items: OrderItem[]
  totalAmount: number
  deliveryAddress: string
  status: OrderStatus
  placedAt: string
  originalAmount: number
  discountAmount: number
  payableAmount: number
  usedVoucher?: Voucher
  merchantDiscountAmount?: number
  platformDiscountAmount?: number
  merchantReceivableAmount?: number
  appliedPromotions?: Promotion[]
  priceSnapshot?: OrderPriceSnapshot | null
  priceBreakdown?: OrderPriceBreakdown | null
  pointsAwarded: number
  refundStatus?: RefundStatus | null
  refundReason?: string | null
  refundImageUrl?: string | null
  refundRequestedAt?: string | null
  refundMerchantReason?: string | null
  refundMerchantReviewedAt?: string | null
  refundAdminReason?: string | null
  refundedAt?: string | null
  customerNoteText?: string | null
  customerNoteImageUrl?: string | null
  statusTimeline?: OrderTimelineEvent[]
  estimatedPrepMinutes?: number | null
  estimatedReadyAt?: string | null
  prepDelayReason?: string | null
  prepDelayedAt?: string | null
  prepTimeoutNotifiedAt?: string | null
}

File: backend/src/order/objects/Order.scala
package delivery.order.objects

import delivery.shared.objects.{MerchantId, OrderId, OrderStatus, RefundStatus, RiderId, UserId, Voucher}

final case class Order(
    id: OrderId,
    customerId: UserId,
    customerName: String,
    customerPhone: String,
    merchantId: MerchantId,
    riderId: Option[RiderId],
    items: List[OrderItem],
    totalAmount: Double,
    deliveryAddress: String,
    status: OrderStatus,
    placedAt: String,
    originalAmount: Double = 0,
    discountAmount: Double = 0,
    payableAmount: Double = 0,
    usedVoucher: Option[Voucher] = None,
    pointsAwarded: Int = 0,
    refundStatus: Option[RefundStatus] = None,
    refundReason: Option[String] = None,
    refundImageUrl: Option[String] = None,
    refundAdminReason: Option[String] = None,
    refundedAt: Option[String] = None,
    customerNoteText: Option[String] = None,
    customerNoteImageUrl: Option[String] = None
)
